### PR TITLE
docs: support inline markdown

### PR DESCRIPTION
**Pull request by lewis6991**
- Tags are now created with `[tag]()`
    - References are now created with `[tag]`
    - Code spans are no longer wrapped

### CODE REVIEW

File: runtime/doc/api.txt [hunk 1]
```diff
@@ -1523,7 +1523,7 @@ nvim_set_keymap({mode}, {lhs}, {rhs}, {opts})              *nvim_set_keymap()*
     To set a buffer-local mapping, use |nvim_buf_set_keymap()|.
 
     Unlike |:map|, leading/trailing whitespace is accepted as part of the
-    {lhs} or {rhs}. Empty {rhs} is |<Nop>|. |keycodes| are replaced as usual.
+    {lhs} or {rhs}. Empty {rhs} is <Nop>. |keycodes| are replaced as usual.
```
Comment (clason) [inline runtime/doc/api.txt:1526]: Link may have been deliberate?

File: runtime/doc/api.txt [hunk 2]
```diff
@@ -3474,9 +3482,9 @@ nvim_create_autocmd({event}, {opts})                   *nvim_create_autocmd()*
                    • event: (string) name of the triggered event
                      |autocmd-events|
                    • group: (number|nil) autocommand group id, if any
-                   • match: (string) expanded value of |<amatch>|
-                   • buf: (number) expanded value of |<abuf>|
-                   • file: (string) expanded value of |<afile>|
+                   • match: (string) expanded value of <amatch>
```
Comment (clason) [inline runtime/doc/api.txt:3485]: Same here

File: runtime/doc/lsp.txt [hunk 3]
```diff
@@ -1002,7 +1001,7 @@ Lua module: vim.lsp.client                                        *lsp-client*
                                 additional checking. If {handler} is not
                                 specified and if there's no respective global
                                 handler, then an error will occur. Returns:
-                                {status}, {[client_id]}. {status} is a boolean
+                                {status}, {client_id}?. {status} is a boolean
```
Comment (clason) [inline runtime/doc/lsp.txt:1004]: 💯

File: runtime/doc/treesitter.txt [hunk 4]
```diff
@@ -820,7 +820,7 @@ inspect_tree({opts})                           *vim.treesitter.inspect_tree()*
 
     While in the window, press "a" to toggle display of anonymous nodes, "I"
     to toggle the display of the source language of each node, "o" to toggle
-    the query editor, and press <Enter> to jump to the node under the cursor
+    the query editor, and press `<Enter>` to jump to the node under the cursor
```
Comment (clason) [inline runtime/doc/treesitter.txt:823]: Keycodes should not be code, right?

File: runtime/doc/api.txt (hunk 1 above)
Comment (lewis6991) [reply runtime/doc/api.txt:1526]: I think redundant, since afaik, keycodes are already links.

File: runtime/doc/treesitter.txt (hunk 4 above)
Comment (lewis6991) [reply runtime/doc/treesitter.txt:823]: Yeah I need to revert this.

File: runtime/doc/api.txt (hunk 1 above)
Comment (clason) [reply runtime/doc/api.txt:1526]: Yeah, that should be changed either way ;)

**Comment by lewis6991** [comment]
Did minimum change. Can replace all the `|foo|` -> `[foo]` in a follow-up

**Pull request merged**